Performance Improvements

kthare10 opened the pull request:
- Fixed indexing for Postgres
- Optimized `FimHelper::update_node()` to access components and interfaces more efficiently
- Updated `fim` version with index fix improving the ASM update time from 12 seconds to few milliseconds
- Modified `Controller:tick_handler` to process the `close` for expiring slivers and `redeem` for ticketed slivers in a thread pool thus improving the over all processing time for a `TickEvent`. Improved from 839 seconds to 30 seconds in worst case.
- Turned lot of INFO logs to DEBUG logs
- Disable the garbage collection from TimerThread - Will be moved to `db_cleanup.py` which runs as a cron job.

ibaldin reviewed (approved):
+1 
This thing should now be flying...

kthare10 commented:
Additional Changes made:
- Modified `PsqlDatabase` class to reuse the per thread session object instead of creating session object per operation
- Modified `orchestrator` `docker-compose.yml` to pass the `NEO4J_dbms_memory_pagecache_size=8G`
- Orchestrator Reservation `ReservationClient` had a bug where the policy member variable was not indicated to be not stored on a pickle. This was creating a huge bottleneck for pickle and db writes.
- Added some more time measurements to for identifying the bottlenecks
- Created a separate database object for REST interface queries
- Removed locking from `ActorDatabase` and the child classes, locking is now handled by per thread `scoped_session` in `PsqlDatabase`

Test Observations:
- 20 slices came up in 18 minutes. 18 minutes depicts the total time but slices were become as soon as all slivers provisioned
- For this test, I did remove all of mine/Paul's dead slices before starting the test
- No congestion observed on the REST interface either
